parsers/bbcontactsparser.py

```diff
@@ -8,14 +8,17 @@ def BbcontactsParser(input):
 
     for line in contents:
         line = line.lstrip().split()
-        if not line or line[1].isalpha():
+        if not line or line[0].isalpha():
             continue
-        elif line[6].isdigit() and line[7].isdigit() and len(line) >= 5:
-            if abs(int(line[6]) - int(line[7])) >= 5:
-                output.append((int(line[6]), int(line[7]), (0)))
+        try:
+            if line[6].isdigit() and line[7].isdigit() and len(line) >= 5:
+                if abs(int(line[6]) - int(line[7])) >= 5:
+                    output.append((int(line[6]), int(line[7]), (0)))
+        except IndexError:
+            raise InvalidFormat('Unable to parse contacts')
```

Comment:
Using `try/except` blocks should almost always be avoided, it's not good practice to assume things will break. Just check that the length of the line is correct before attempting to parse it... Instead of `len(line) >= 5` why don't you check `len(line) >= 8`?

Reply:
Also, bear in mind that with python you can short-circuit the evaluation of conditional statements. This is useful when checking multiple conditions that could cause problems if one of them fails. For example, what you have now is:

`if line[6].isdigit() and line[7].isdigit() and len(line) >= 5:`

If the list `line` is smaller than 5, it will throw an index error when checking `if line[6].isdigit() `. However, if you change the order of the conditions and write this:

`if len(line) >= 8 and line[6].isdigit() and line[7].isdigit()`

If `len(line) >= 8` does not return `True`, python will short-circuit the `if` statement and not test whatever conditions you put afterwards, it will never try to check `if line[6].isdigit() and line[7].isdigit()` and therefore it will never raise `IndexError` when trying to access an element outside the list.